pkg/scheduler/scheduler.go

```diff
@@ -253,6 +252,26 @@ func New(client clientset.Interface,
 	for _, opt := range opts {
 		opt(&options)
 	}
+
+	schedulerCache := internalcache.New(30*time.Second, stopCh)
+	volumeBinder := volumebinder.NewVolumeBinder(client, nodeInformer, pvcInformer, pvInformer, storageClassInformer,
+		time.Duration(options.bindTimeoutSeconds)*time.Second)
+
+	if options.frameworkRegistry == nil {
+		options.frameworkRegistry = frameworkplugins.NewDefaultRegistry(&frameworkplugins.DefaultRegistryArgs{
```

Comment:
If we create a new function(e.g. `NewRegistryArgs`), we could call it from `cmd/kube-scheduler/app/server.go`, then we do not need to disable out-of-tree framework plugins.

Reply:
that will require us to define global variables for the cache and volume binder, which is a pain to deal with. The framework is changing so fast right now there is no point in actually supporting out-of-tree plugins before beta.

Reply:
> that will require us to define global variables for the cache and volume binder

I could not get your point why we need to define global variables for them, we could still declare them as local variables and use them as following. Is there any problem?
```
	schedulerCache := internalcache.New(30*time.Second, stopCh)
	volumeBinder := volumebinder.NewVolumeBinder(client, nodeInformer, pvcInformer, pvInformer, storageClassInformer
```

> The framework is changing so fast right now there is no point in actually supporting out-of-tree plugins before beta.

We are trying this feature, not sure whether others are using it now. Even it is alpha, we could get feedback from users. If it is not a must, I think it might be better to keep it.

Reply:
> > that will require us to define global variables for the cache and volume binder
> 
> I could not get your point why we need to define global variables for them, we could still declare them as local variables and use them as following. Is there any problem?
> 
> ```
> 	schedulerCache := internalcache.New(30*time.Second, stopCh)
> 	volumeBinder := volumebinder.NewVolumeBinder(client, nodeInformer, pvcInformer, pvInformer, storageClassInformer
> ```
> 

schedulerCache and volumeBinder are shared handlers, we can't instantiate them multiple times, they need to be created once.

> > The framework is changing so fast right now there is no point in actually supporting out-of-tree plugins before beta.
> 
> We are trying this feature, not sure whether others are using it now. Even it is alpha, we could get feedback from users. If it is not a must, I think it might be better to keep it.

It is going to be tricky to keep this, and it is very premature to say we support out-of-tree plugins at this stage because we are changing the framework very frequently as we discover its limitation during the migration. If you are testing this feature, I think you can still carry a small patch to enable this back in your dev environment. Is that reasonable?